mycli/packages/special/iocommands.py

```diff
@@ -240,3 +240,50 @@ def execute_system_command(arg, **_):
         return [(None, None, None, response)]
     except OSError as e:
         return [(None, None, None, 'OSError: %s' % e.strerror)]
+
+@export
+def set_redirected_output(append=False, filename=None, enabled=True):
+    global use_redirected_output
+    use_redirected_output = {'append': append, 'filename':filename, 'enabled': enabled}
+
+@export
+def get_redirected_output():
+    return use_redirected_output
+
+@export
+def redirectmatch(sql):
+    filename = None
+    if sql[-1] in ("'", '"'): # find filename in quoted redirect
+        quote = sql[-1]
+        m = re.match(r'.*([^\\]'+quote+r'.*?'+quote+'$)', sql)
```

Comment:
Rather than putting the parenthesis around the quotes and the filename, put them just around the filename (inside the quotes), so that you don't have to remove the quotes later on. Then, `filename = m.group(1)` won't include the quotes.

Then you can get rid of line 262 (`quote = ''`) and lines 277-278 (`if quote: filename = filename[len(quote):-len(quote)]`).

Side note: if possible, put spaces around operators like `+`, e.g. `re.match(foo + bar)` instead of `re.match(foo+bar)`. It makes it easier to read.